cellengine/resources/attachment.py

```diff
@@ -1,58 +1,49 @@
 from __future__ import annotations
-from dataclasses import dataclass, field
-from dataclasses_json.cfg import config
+
+from attr import define, field
 
 import cellengine as ce
-from cellengine.utils.dataclass_mixin import DataClassMixin, ReadOnly
+from cellengine.utils import converter, readonly
 
 
-@dataclass
-class Attachment(DataClassMixin):
+@define
+class Attachment:
     """A class representing a CellEngine attachment.
     Attachments are non-data files that are stored in an experiment.
     """
 
+    _id: str = field(on_setattr=readonly)
     filename: str
-    _id: str = field(
-        metadata=config(field_name="_id"), default=ReadOnly()
-    )  # type: ignore
-    crc32c: str = field(repr=False, default=ReadOnly())  # type: ignore
-    experiment_id: str = field(repr=False, default=ReadOnly())  # type: ignore
-    md5: str = field(repr=False, default=ReadOnly())  # type: ignore
-    size: int = field(repr=False, default=ReadOnly())  # type: ignore
+    crc32c: str = field(on_setattr=readonly, repr=False)
+    experiment_id: str = field(on_setattr=readonly, repr=False)
+    md5: str = field(on_setattr=readonly, repr=False)
+    size: int = field(on_setattr=readonly, repr=False)
 
-    @classmethod
-    def get(cls, experiment_id: str, _id: str = None, name: str = None) -> Attachment:
-        """Get an Attachment by name or ID for a specific experiment. Either
-        `name` or `_id` must be specified.
+    @property
+    def client(self):
+        return ce.APIClient()
 
-        Args:
-            experiment_id: ID of the experiment this attachment is connected with.
-            _id (optional): ID of the attachment.
-            name (optional): Name of the experiment.
-        """
-        kwargs = {"name": name} if name else {"_id": _id}
-        return ce.APIClient().get_attachment(experiment_id, **kwargs)
-
-    @staticmethod
-    def upload(experiment_id: str, filepath: str, filename: str = None) -> Attachment:
```

Comment:
What happened to this method too?

Reply:
Also replaced by `client.upload_attachment()`; same as above, if you don't like this I'll keep it. I just found that re-implementing `client` methods in other classes seemed a little convoluted; since the user must use the `client` for several critical pieces anyway, we may as well encourage them to do so.

Reply:
Authentication is the biggest, certainly, and `get_experiment` or `get_experiments` is typically the first method a user will use. From there, the user _could_ do most things from the `Experiment`, but I _think_ the client is the best entrypoint for using toolkit. If you (or Olivier) think otherwise after using it a bit, I'm happy to change the direction I'm pushing the user :smiley_cat:

Reply:
I don't think we need to provide so many ways for users to do things. I'd also encourage the class methods, including `Experiment.get` and `Experiment.list`. They're conceptually organized.

Reply:
The only thing is that `Experiment.get`, for instance, just calls `APIClient.get_experiment`. My motivation here is to have there be fewer ways of doing things, although I also like the conceptual organization of `<Entity>.get`, `.update`, `.create`, `.delete`